synapse/events/spamcheck.py

```diff
@@ -71,6 +71,11 @@
         Union[
             Literal["NOT_SPAM"],
             "synapse.api.errors.Codes",
+            # Highly experimental, not officially part of the spamchecker API, may
+            # disappear without warning depending on the results of ongoing
+            # experiments.
+            # Use this to return additional information as part of an error.
+            Tuple["synapse.api.errors.Codes", Dict],
```

Comment:
I'd prefer if we imported `Codes` directly and used it here. I appreciate that other bits of of this file are referring to `Codes` in that way here and it's probably my bad for missing that in previous PRs, I'd say it's probably best to get those in line at the same time.

Reply:
I have been bitten by circular dependencies around `synapse.api.errors.Codes`. I seem to remember that this is why I refer to it by its full path instead of importing it.

Reply:
I might be starting to jump at shadows :)
I'm a tad nervous about how easy to is to break Python code with innocuous changes.